terraform/modules/network/main.tf

```diff
@@ -0,0 +1,161 @@
+################################################################################
+# VPC Definition
+################################################################################
+resource "aws_vpc" "this" {
+  cidr_block                       = var.vpc_cidr
+  assign_generated_ipv6_cidr_block = true
+  enable_dns_support               = var.enable_dns_support
+  enable_dns_hostnames             = var.enable_dns_hostnames
+
+  tags = {
+    Name = "eda-vpc"
+  }
+}
+
+################################################################################
+# Availability zones
+################################################################################
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+################################################################################
+# IG Definition
+################################################################################
+resource "aws_internet_gateway" "igw" {
+  vpc_id = aws_vpc.this.id
```

Comment:
Hm this way we always create a IGW compared to previous version where you could select if you want to create or not.

Reply:
Yes true but for our usecase we always want an IGW etc. I think we should not make the modules fully customisable as this are only private modules for our project and not public ones. Else we could directly use the already available modules from terraform